### PR TITLE
feat: introduce `--Xfuzzer`

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@herluf-ba This seems useful. Just running it a few times I hit stuff like:

```
Exception in thread "main" java.lang.AssertionError: assertion failed
	at scala.Predef$.assert(Predef.scala:264)
	at ca.uwaterloo.flix.language.phase.Parser2$Expr$.withRule(Parser2.scala:2020)
```

and 

```
ca.uwaterloo.flix.util.InternalCompilerException: [BigDecimal.flix:185:46] Parser is stuck
	at ca.uwaterloo.flix.language.phase.Parser2$.ca$uwaterloo$flix$language$phase$Parser2$$nth(Parser2.scala:323)
```

File: main/src/ca/uwaterloo/flix/language/phase/Lexer.scala [hunk 1]
```diff
@@ -1089,4 +1090,35 @@ object Lexer {
...
+      return tokens
+    }
+
+    // Return immediately if the array of tokens is too small.
+    if (tokens.length <= 10) {
+      return tokens
+    }
+
+    //
+    // We fuzz the array of tokens by picking two random indices and swapping the,
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Lexer.scala:1111]: swapping the what?

File: main/src/ca/uwaterloo/flix/language/phase/Lexer.scala [hunk 2]
```diff
@@ -1089,4 +1090,35 @@ object Lexer {
     }
     TokenKind.Err(LexerError.UnterminatedBlockComment(sourceLocationAtStart()))
   }
+
+  /**
+    * Returns a new array of tokens based on the successfully lexed array of `tokens`.
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/phase/Lexer.scala:1095]: Doesn't actually return a new array, right? Modifies the given array and returns it.

**Review by mlutze — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Peer programming with a baby--less useful than you would think.

**Pull request merged**